pkgs/applications/networking/browsers/firefox/packages.nix

```diff
@@ -120,85 +62,21 @@ rec {
-
-        - entirely free software, no non-free plugins, addons,
-          artwork,
-        - no telemetry, no "studies",
-        - sane privacy and security defaults (for instance, unlike
-          Firefox, IceCat does _zero_ network requests on startup by
-          default, which means that with IceCat you won't need to
-          unplug your Ethernet cable each time you want to create a
-          new browser profile without announcing that action to a
-          bunch of data-hungry corporations),
-        - all essential privacy and security settings can be
-          configured directly from the main screen,
-        - optional first party isolation (like TorBrowser),
-        - comes with HTTPS Everywhere (like TorBrowser), Tor Browser
-          Button (like TorBrowser Bundle), LibreJS, and SpyBlock
-          plugins out of the box.
-
-        This package can be installed together with Firefox and
-        TorBrowser, it will use distinct binary names and profile
-        directories.
-      '';
-      homepage = "https://www.gnu.org/software/gnuzilla/";
-      platforms = lib.platforms.unix;
-      license = with lib.licenses; [ mpl20 gpl3Plus ];
-    };
-  });
-
-in {
-
-  icecat = iccommon rec {
-    ffversion = "60.3.0";
-    icversion = "${ffversion}-gnu1";
-
-    src = fetchurl {
-      url = "mirror://gnu/gnuzilla/${ffversion}/icecat-${icversion}.tar.bz2";
-      sha256 = "0icnl64nxcyf7dprpdpygxhabsvyhps8c3ixysj9bcdlj9q34ib1";
-    };
-
-    patches = [
-      ./no-buildconfig.patch
-      missing-documentation-patch
-    ];
-    meta.knownVulnerabilities = [ "Support ended around October 2019." ];
-  };
-
-  # Similarly to firefox-esr-52 above.
-  icecat-52 = iccommon rec {
-    ffversion = "52.6.0";
-    icversion = "${ffversion}-gnu1";
 
-    src = fetchurl {
-      url = "mirror://gnu/gnuzilla/${ffversion}/icecat-${icversion}.tar.bz2";
-      sha256 = "09fn54glqg1aa93hnz5zdcy07cps09dbni2b4200azh6nang630a";
-    };
+  icecat = throw "firefoxPackages.icecat was removed as even it's latest upstream version is based on an unsupported ESR release with open security issues.";
 
-    patches = [
-      # this one is actually an omnipresent bug
-      # https://bugzilla.mozilla.org/show_bug.cgi?id=1444519
-      ./fix-pa-context-connect-retval.patch
-    ];
+  icecat-52 = throw "firefoxPackages.icecat was removed as even it's latest upstream version is based on an unsupported ESR release with open security issues.";
 
-    meta.knownVulnerabilities = [ "Support ended in August 2018." ];
-  };
+  firefox-esr-52 = throw ''
+    firefoxPackages.firefox-esr-52 was removed as it's an unsupported ESR with
+    open security issues. If you need it because you need to run some plugins
+    not having been ported to WebExtensions API, import it from an older
+    nixpkgs checkout still containing it.
+  '';
+  firefox-esr-60 = throw "firefoxPackages.firefox-esr-60 was removed as it's an unsupported ESR with open security issues.";
 
```

Comment:
we need to merge these aliases like https://github.com/NixOS/nixpkgs/blob/d4a951f31d2d6a542f2fa6387df2a269f6b9e042/pkgs/desktops/xfce/default.nix#L154.
same to how it is in `aliases.nix`

Reply:
fixed.

Reply:
I tried, but this now breaks eval, as some of these are referenced by top-level, too.

I don't really want to add another list of aliases and am inclided to drop the last commit introducing the special alias handling.

Reply:
Ah, this is because
```
all-packages.nix
ficecat-unwrapped = firefoxPackages.icecat;
icecat = wrapFirefox icecat-unwrapped { };
```

@flokli You have to remove the `firefox-esr-60`, `icecat-52`, and `icecat` attributes in `all-packages.nix` too.

What you can do is keep the `throws` exactly as they are here, but do `aliases` in `aliases.nix` to bind
```
icecat-unwrapped = icecat;
icecat = firefoxPackages.icecat; # now everyone gets the throw
```

and mark to remove them next release.

Reply:
done.